module/data/item/weapon.mjs

```diff
@@ -285,6 +285,23 @@ export default class WeaponData extends ItemDataModel.mixin(
   /*  Properties                                  */
   /* -------------------------------------------- */
 
+  /**
+   * Ammunition that can be used with this weapon.
+   * @type {FormSelectOption[]}
+   */
+  get ammunitionOptions() {
+    if ( !this.parent.actor ) return [];
+    return this.parent.actor.items
+      .filter(i => (i.type === "consumable") && (i.system.type.value === "ammo")
+        && (!this.ammunition?.type || (i.system.type.subtype === this.ammunition.type)))
+      .map(i => ({
+        value: i.id, label: `${i.name} (${i.system.quantity})`, item: i, disabled: !i.system.quantity
+      }))
+      .sort((lhs, rhs) => lhs.label.localeCompare(rhs.label, game.i18n.lang));
```

Comment:
`itemTypes` is cached now so it saves us iterating over the entire items collection here:

```suggestion
    return this.parent.actor.itemTypes.consumable
      .filter(i => (i.system.type.value === "ammo")
        && (!this.ammunition?.type || (i.system.type.subtype === this.ammunition.type)))
      .map(i => ({
        value: i.id, label: `${i.name} (${i.system.quantity})`, item: i, disabled: !i.system.quantity
      }))
      .sort((lhs, rhs) => lhs.label.localeCompare(rhs.label, game.i18n.lang));
```